editor/filesystem_dock.cpp

```diff
@@ -1543,6 +1546,11 @@ void FileSystemDock::_file_multi_selected(int p_index, bool p_selected) {
 
 void FileSystemDock::_file_selected() {
 
+	file_selected = true;
+	set_process(true);
```

Comment:
It took me some time to understand why you're doing this, but I see now that it's to defer the `_update_import_dock` call so that it happens only once thanks to the bool. Still, it feels a bit hackish to use processing for that.

How about instead:
```cpp
void FileSystemDock::_file_multi_selected() {
    import_dock_needs_update = true;
    call_deferred("_update_import_dock");
}

void FileSystemDock::_file_selected() {
    import_dock_needs_update = true;
    _update_import_dock();
}

void FileSystemDock::_update_import_dock() {
    if (!import_dock_needs_update)
        return;
[...]
    import_dock_needs_update = false;
}
```

Though maybe it's still as hackish, I don't know :P

Reply:
yeap. it's a bit hackish but couldn't find a better way without touching `ItemList`
I will try it in that way too.

Reply:
@akien-mga your suggestion works as expected also, and I think it's better.
so I updated PR. :)